### ISSUE TITLE
cfn-lint shows `W1020 Fn::Sub isn't needed` on Serverless LayerVersion property LayerName when a Variable is actually present

### ISSUE THREAD

**Issue by patrickgreenwell** [aws-cloudformation/cfn-python-lint#897]
*cfn-lint version: 0.20.1*

cfn-lint shows `W1020 Fn::Sub isn't needed` on Serverless LayerVersion property LayerName when a Variable is actually present in the layerName String of the intrinsic function Sub

Please provide as much information as possible:

sample resource: 
```yaml
APICommonCodeLayer:
    Type: 'AWS::Serverless::LayerVersion'
    Properties:
      Description: API common libraries
      LayerName: 
        Fn::Sub: 'api-common-code-layer-${Environment}'
      ContentUri: api/common
      CompatibleRuntimes:
        - nodejs8.10
      RetentionPolicy: Retain
```
Where Environment is a Paramater with the following definition:

```yaml
Environment:
    AllowedValues:
      - Dev
      - Qa
      - Staging
      - Mirror
      - Prod
    ConstraintDescription: You must specify a valid EnvironmentValue
    Default: Dev
    Description: Environment for to Deploy to
    Type: String
```
This only seems to affect LayerVersions as other resources proceed to not return the warning when using Fn::Sub in their names, even other AWS::Serverless resources.

Sample Template:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'
Description: SampleTemplate
Parameters:
  Environment:
    Type: String
    AllowedValues:
      - Dev
      - Qa
      - Staging
      - Mirror
      - Prod
    ConstraintDescription: You must specify a valid EnvironmentValue
    Default: Dev
    Description: Environment for to Deploy to
Resources:
  APICommonCodeLayer:
    Type: 'AWS::Serverless::LayerVersion'
    Properties:
      Description: API common libraries
      LayerName: 
        Fn::Sub: 'api-common-code-layer-${Environment}'
      ContentUri: api/common
      CompatibleRuntimes:
        - nodejs8.10
      RetentionPolicy: Retain
```

**Comment by vancleimatheus** [comment]
I believe this bug is back on the newer version of  `cfn-lint`.

`            Resource:  !Sub '{{resolve:ssm:/mock/lambda/my-resource/arn}}'`

The line above work just fine  on previous version but with the latest version it gives me:

`'Fn::Sub' isn't needed because there are no variables`

**Comment by kddejong** [comment]
> I believe this bug is back on the newer version of `cfn-lint`.
> 
> ` Resource: !Sub '{{resolve:ssm:/mock/lambda/my-resource/arn}}'`
> 
> The line above work just fine on previous version but with the latest version it gives me:
> 
> `'Fn::Sub' isn't needed because there are no variables`

Fix in v1.6.0

**Comment by vancleimatheus** [comment]
Awesome, it works now, thank you

**Comment by kddejong** [comment]
@pjlowrey looking into this.

**Comment by kddejong** [comment]
@pjlowrey may need some more details on this one.  I can't replicate it.

The first thing this [rule](https://github.com/aws-cloudformation/cfn-lint/blob/0dd93b2671f728d6b519eeac77d6ba2f148f605b/src/cfnlint/rules/functions/SubUnneeded.py#L27-L30) does is look for the SAM transform. We look at the pre transform (value of Transform prior to any transforms) as SAM will remove the transform section on completion.

Any chance you are using multiple Transforms?  Language Extension for instance?

**Comment by pjlowrey** [comment]
@kddejong I am only using the single `Transform: AWS::Serverless-2016-10-31` and no Language Extension. 

The line cfn-lint is erroring on is `FunctionArn: !Sub arn:aws:lambda:us-east-1:123456789012:function:Lambda_Function_${env}` and I do have an 'env' parameter defined above in the template.

I'm not super familiar with much of this, let me know what helpful context I may be able to provide.
